### PR TITLE
add is_half argument to extract_feature_print.py

### PR DESCRIPTION
**Pull request by Zzzzzzyt**
# PR type

- Feature / Bug fix: add is_half argument to extract_feature_print.py

# Description

Adds commandline argument is_half to extract_feature_print.py as the new last argument.
Fixes the problem of generating NaN when run on certain GPUs that lack proper fp16 support (e.g. my GTX1650)

Might break 3rd-party scripts.

# Screenshot

An example output of the bug in action:

```['infer/modules/train/extract_feature_print.py', 'cuda:0', '1', '0', '0', 'D:\\MyPython\\Retrieval-based-Voice-Conversion-WebUI/logs/mi-test', 'v2']
D:\MyPython\Retrieval-based-Voice-Conversion-WebUI/logs/mi-test
load model(s) from assets/hubert/hubert_base.pt
move model to cuda
all-feature-17
0_17.wav-contains nan
now-17,all-6,0_17.wav,(184, 768)
0_18.wav-contains nan
now-17,all-7,0_18.wav,(184, 768)
0_2.wav-contains nan
now-17,all-8,0_2.wav,(184, 768)
0_20.wav-contains nan
now-17,all-9,0_20.wav,(184, 768)
0_21.wav-contains nan
now-17,all-10,0_21.wav,(184, 768)
0_23.wav-contains nan
now-17,all-11,0_23.wav,(184, 768)
0_25.wav-contains nan
now-17,all-12,0_25.wav,(53, 768)
0_3.wav-contains nan
now-17,all-13,0_3.wav,(184, 768)
0_5.wav-contains nan
now-17,all-14,0_5.wav,(184, 768)
0_7.wav-contains nan
now-17,all-15,0_7.wav,(124, 768)
0_8.wav-contains nan
now-17,all-16,0_8.wav,(184, 768)
all-feature-done
```

### CODE REVIEW

File: infer/modules/train/extract_feature_print.py [hunk 1]
```diff
@@ -91,8 +93,9 @@ def readwave(wav_path, normalize=False):
 model = models[0]
 model = model.to(device)
 printt("move model to %s" % device)
-if device not in ["mps", "cpu"]:
-    model = model.half()
+if is_half:
+    if device not in ["mps", "cpu"]:
+        model = model.half()
```
Comment (Tps-F) [inline infer/modules/train/extract_feature_print.py:98]: mps and cpu need to use half

**Review by Tps-F — CHANGES_REQUESTED** [review]

File: infer/modules/train/extract_feature_print.py (hunk 1 above)
Comment (Zzzzzzyt) [reply infer/modules/train/extract_feature_print.py:98]: I don't quite understand why this is the case
Judging from how config.py is written mps and cpu need to *not* use half

https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/0de65053247fb616fa696e4c40d32411174b6310/configs/config.py#L168-L177

I have no experience working with mps though

Comment (Tps-F) [reply infer/modules/train/extract_feature_print.py:98]: Oh really, What was I thinking?
I rewroted this config.py and also the configuration for mps

**Comment by Tps-F** [comment]
Have you checked the `extract_feature_print.py` on devices that have not been half before?

**Comment by Zzzzzzyt** [comment]
This pull request will not change behavior when device is cpu or mps, they _always have been_ full precision
As for fp32 vs fp16 on cuda unfortunately I only have GTX1650 on hand so I cannot test on proper fp16 gpus (since RTX20xx)

**Comment by Tps-F** [comment]
> This pull request will not change behavior when device is cpu or mps, they always have been full precision

Yeah, i know

> As for fp32 vs fp16 on cuda unfortunately I only have GTX1650 on hand so I cannot test on proper fp16 gpus (since RTX20xx)

It’s okay, thanks.

**Review by fumiama — APPROVED** [review]

**Comment by jie-chen** [comment]
Hi, in Line 129 https://github.com/Zzzzzzyt/Retrieval-based-Voice-Conversion-WebUI/blob/ab803bc4718387b4eb0ac00fc456b4f2c9b19494/infer/modules/train/extract_feature_print.py#L129 Features are still stored as fp32. Is this intentional?

**Pull request merged**